### PR TITLE
Spring Exercise 5 complete

### PR DESCRIPTION
**Pull request by asish918**
# Homework 9 Asish

### Topics Covered:
- Validations and Error handling
- Spring Profiles

### Drift Video Explanation
[Drift](https://app.drift.com/video/share/abO4WOd0wN3u2fMoeY1Z6mozCNzr2rb8mK892w420I)

### Output Screenshots - 
Screenshot 2024-01-25 at 23 43 22

Screenshot 2024-01-25 at 23 43 08

Screenshot 2024-01-25 at 23 43 48

Screenshot 2024-01-25 at 23 43 45

Screenshot 2024-01-25 at 23 43 42

Screenshot 2024-01-25 at 23 43 36

### The code smells are unused imports which I forgot to remove. Please excuse them 🙃

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 3 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

File: backend/homework/9/src/main/java/asish/kdu/spring_exercise_4/controller/AppController.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * Endpoint to find a book by its name
+     * @param name Name of the book
+     * @return Response with all the Book details
+     * @throws BookException When book cant be found
+     */
+    @GetMapping("/book")
+    public ResponseEntity<BookDTO> getBook(@RequestParam String name) throws BookException {
+        BookDTO res = bookService.getBookByName(name);
```
Comment (rishav-kd) [inline backend/homework/9/src/main/java/asish/kdu/spring_exercise_4/controller/AppController.java:48]: Let the service function return ResponseEntity, Incase the function returns an exception the response can still be 200 ok

File: backend/homework/9/src/main/java/asish/kdu/spring_exercise_4/logging/CustomLogger.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * CustomLogger has been defined with support for various levels of logging based on enum input
+ */
+public class CustomLogger {
+    public enum LoggerType {
+        DEBUG,
+        INFO,
+        WARNING,
+        ERROR,
+    }
```
Comment (rishav-kd) [inline backend/homework/9/src/main/java/asish/kdu/spring_exercise_4/logging/CustomLogger.java:14]: lets have this in a separate file.

**Review by rishav-kd — COMMENTED** [review]
Looks great! Only nitpicking - try to use negative usecases in your code for drift video

**Pull request closed without merging**